style the invite pages

staltz opened the pull request:
# Desktop

## Invite created

![invite-created-desktop](https://user-images.githubusercontent.com/90512/113011401-6fa82a80-9182-11eb-9b50-8d7cb1b33059.png)

## Join

![join-default-desktop](https://user-images.githubusercontent.com/90512/113011454-7df64680-9182-11eb-842d-f93736ccfa9b.png)

## Join fallback

![join-fallback-desktop](https://user-images.githubusercontent.com/90512/113011479-851d5480-9182-11eb-9d0f-a4cd87d10834.png)

## Join manually

![join-manually-desktop](https://user-images.githubusercontent.com/90512/113011512-8b133580-9182-11eb-9554-511fd74279a0.png)

## Join manually succeeded

![joined-desktop](https://user-images.githubusercontent.com/90512/113011577-98c8bb00-9182-11eb-9bf9-911ae3f4ddad.png)

<details>
<summary>Mobile</summary>

## Invite created

![invite-created-mobile](https://user-images.githubusercontent.com/90512/113011636-a7af6d80-9182-11eb-918e-a553fe37381d.png)

## Join

![join-default-mobile](https://user-images.githubusercontent.com/90512/113011658-ac742180-9182-11eb-907c-9e9b23e39764.png)

## Join fallback

![join-fallback-mobile](https://user-images.githubusercontent.com/90512/113011677-b0a03f00-9182-11eb-9246-dfb609e0144b.png)

## Join manually

![join-manually-mobile](https://user-images.githubusercontent.com/90512/113011695-b433c600-9182-11eb-8a42-6288c04ca66d.png)

## Join manually succeeded

![joined-mobile](https://user-images.githubusercontent.com/90512/113011702-b72eb680-9182-11eb-91e4-b678e8c405af.png)

</details>

staltz commented:
Alright, I'll review the tests that I broke completely.

staltz commented:
@cryptix This is ready for review :)

cryptix commented on web/handlers/invites.go:
```diff
@@ -60,13 +54,88 @@ func (h inviteHandler) presentFacade(rw http.ResponseWriter, req *http.Request)
 
 	joinRoomURI.RawQuery = queryVals.Encode()
 
+	return joinRoomURI.String()
```
```suggestion
	return template.URL(joinRoomURI.String())
```

cryptix reviewed (approved):
Small nitpick on the `buildJoinRoomURI` function but looks all good from my end.

I just eyeballed the template changes, tho.

Big +1 for the docs folder!

staltz commented:
@cryptix Thanks for the suggested changes, I incorporated them now. Also, I added the chart.graphml file, which is the source file in case we need to update that later.

cryptix commented:
Ah... that's what you get for making code changes on github..


> web/handlers/invites.go:82:27: cannot use joinRoomURI (type "html/template".URL) as type string in argument to qrcode.New

https://github.com/ssb-ngi-pointer/go-ssb-room/runs/2277216659#step:6:7

staltz commented:
Yeah I was looking into now, not sure how to get a string from that.

cryptix commented on web/handlers/invites.go:
```diff
@@ -60,13 +54,88 @@ func (h inviteHandler) presentFacade(rw http.ResponseWriter, req *http.Request)
+	fallbackURL := urlTo(router.CompleteInviteFacadeFallback, "token", token)
+
+	// generate a QR code with the token inside so that you can open it easily in a supporting mobile app
+	qrCode, err := qrcode.New(joinRoomURI, qrcode.Medium)
```
```suggestion
	qrCode, err := qrcode.New(string(joinRoomURI), qrcode.Medium)
````

this should do the trick.

cryptix commented:
> not sure how to get a string from that.

In the case where it's a _type aliases_ (i.e. `type Foo int`) you can do this:

```go
type Foo int

var i = 42
var asF = Foo(i)

var asIntAgain = int(asF)
```

Same for `bool` or `string`. It's just for complex things like structs where it get's hairy but the compiler will yell at you also if you'd try.

staltz commented:
Thanks! I'll let you press the big green button
